crates/resolver/src/tls/dns_over_rustls.rs

```diff
@@ -71,3 +73,26 @@ pub(crate) fn new_tls_stream<R: RuntimeProvider>(
         tls_client_connect_with_bind_addr(socket_addr, bind_addr, dns_name, client_config);
     (Box::pin(stream), handle)
 }
+
+#[allow(clippy::type_complexity)]
+pub(crate) fn new_tls_stream_with_future<S, F>(
```

Comment:
Why do we need this extra layer of abstraction?

Reply:
I didn't get what you mean. Could you make some explanation?

Reply:
It's not clear to me why we need this extra function.

Reply:
This is the replacement of `new_tls_stream`. I think I should remove the former one since it's not used at all. If you want to know why the original function is needed, it's because we have rustls, native-tls and openssl implementations.

Reply:
Right, if we can remove `new_tls_stream()` because it's gone unused we should definitely do that.